src/integrationtests/java/com/aws/greengrass/integrationtests/ConfigTest.java

```diff
@@ -78,6 +87,90 @@ void GIVEN_Greengrass_with_mqtt_bridge_WHEN_multiple_config_changes_consecutivel
+                .payloadFormat(Publish.PayloadFormatIndicator.UTF8)
+                .contentType("contentType")
+                .build();
+
+        Consumer<MqttMessage> messageHandler = (message) -> {
+          assertEquals(Arrays.toString(expectedMessage.getPayload()), Arrays.toString(message.getPayload()));
+        };
+
+        testContext.getLocalV5Client().updateSubscriptions(topics, messageHandler);
+        testContext.getLocalV5Client().publish(
+                MqttMessage.builder()
+                        .topic(topic)
+                        .payload("abc".getBytes(StandardCharsets.UTF_8))
+                        .userProperties(Collections.singletonList(new UserProperty("key", "val")))
+                        .responseTopic("response topic")
+                        .messageExpiryIntervalSeconds(1234L)
+                        .payloadFormat(Publish.PayloadFormatIndicator.UTF8)
+                        .contentType("contentType")
+                        .build());
+        TimeUnit.MILLISECONDS.sleep(100);
+
+        // change config values
+        config.lookup(BridgeConfig.KEY_SESSION_EXPIRY_INTERVAL).withValue(1);
+        config.lookup(BridgeConfig.KEY_MAXIMUM_PACKET_SIZE).withValue(1);
+        config.lookup(BridgeConfig.KEY_RECEIVE_MAXIMUM).withValue(1);
+
+        // verify that config changes take effect
+        assertThat("session expiry interval config update",
+                () -> testContext.getLocalV5Client().getSessionExpiryInterval(), eventuallyEval(is(1L)));
+        assertThat("maximum packet size config update", () -> testContext.getLocalV5Client().getMaximumPacketSize(),
+                eventuallyEval(is(1L)));
+        assertThat("receive maximum config update", () -> testContext.getLocalV5Client().getReceiveMaximum(),
+                eventuallyEval(is(1)));
+
+        // Publish a large message to IoT Core and verify that it is not received due to the local client's config
+        // We expect the response to be null since the previous message was too large to successfully publish, given
+        // the local client's config
+        Consumer<MqttMessage> largeMessageHandler = Assertions::assertNull;
+        testContext.getLocalV5Client().updateSubscriptions(topics, largeMessageHandler);
+        testContext.getLocalV5Client().publish(
+                MqttMessage.builder()
+                        .topic(topic)
+                        .payload("this message is too large to be published".getBytes(StandardCharsets.UTF_8))
+                        .userProperties(Collections.singletonList(new UserProperty("key", "val")))
+                        .responseTopic("response topic")
+                        .messageExpiryIntervalSeconds(1234L)
+                        .payloadFormat(Publish.PayloadFormatIndicator.UTF8)
+                        .contentType("contentType")
+                        .build());
+        TimeUnit.MILLISECONDS.sleep(100);
```

Comment:
should also wait for the largeMEssageHandler to complete first so we know the message has been published

Reply:
i think i misspoke, we dont have a way to hook into that right now